### PR TITLE
Fix internal breakage caused by #82134

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -350,6 +350,18 @@ def get_global_rank(group: ProcessGroup, group_rank: int) -> int:
             return rank
     raise RuntimeError(f"Group rank {group_rank} is not part of group {group}")
 
+# TODO: remove this once the ecosystem moves away from it.
+def _get_global_rank(group, rank):
+    """
+    This method is deprecated, please use get_global_rank.
+    """
+    warnings.warn(
```
Comment (rohan-varma) [inline torch/distributed/distributed_c10d.py:358]: warn only once to avoid spamming logs maybe?

**Review by rohan-varma — APPROVED** [review]

**Review by mehtanirav — APPROVED** [review]

File: torch/distributed/distributed_c10d.py (hunk 1 above)
Comment (kumpera) [reply torch/distributed/distributed_c10d.py:358]: warnings.warn does that by default

**Pull request closed without merging**